openlibrary/plugins/openlibrary/api.py

```diff
@@ -212,3 +217,35 @@ def get_works_data(self, author, limit, offset):
             "size": size,
             "entries": works
         }
+
+class price_api(delegate.page):
+    path = r'/prices/isbn/(.*)'
+
+    @jsonapi
+    def GET(self, isbn):
+        metadata = {
+            'amazon': get_amazon_metadata(isbn) or {},
+            'betterworldbooks': get_betterworldbooks_metadata(isbn) or {}
+        }
+        if len(isbn) == 10 and \
+           metadata['betterworldbooks'].get('price') is None:
+            isbn_13 = isbn_10_to_isbn_13(isbn)
+            metadata['betterworldbooks'] = get_betterworldbooks_metadata(
+                isbn_13) or {}
+
+        # fetch book by isbn if it exists
+        book = web.ctx.site.things({
+            "type": "/type/edition", 'isbn': isbn
+        })
+
+        # if no OL edition for isbn, attempt to create
```

Comment:
price_api attempts to create OL entry from amazon (if amazon data returned and no ol edition exists for isbn)